python/tach/cache/access.py

```diff
@@ -1,19 +1,47 @@
 from __future__ import annotations
 
+import json
 import uuid
-from pathlib import Path
+from urllib import error, request
 
-from tach.cache.setup import resolve_dot_tach
-from tach.filesystem import find_project_config_root
+from tach.cache.setup import get_project_path, resolve_dot_tach
 
 
 def get_uid() -> uuid.UUID | None:
-    project_root = find_project_config_root(str(Path.cwd()))
-    if project_root is None:
+    project_path = get_project_path()
+    if project_path is None:
         return
-    project_path = Path(project_root)
     if not (project_path / ".tach" / "tach.info").exists():
         resolve_dot_tach()
-    with open(project_path / ".tach" / "tach.info") as f:
-        uid = uuid.UUID(f.read().strip())
+    contents = (project_path / ".tach" / "tach.info").read_text().strip()
+    uid = uuid.UUID(contents)
     return uid
+
+
+def get_latest_version() -> str | None:
+    project_path = get_project_path()
+    if project_path is None:
+        return
+    if not (project_path / ".tach" / ".latest-version").exists():
+        return
+    version = (project_path / ".tach" / ".latest-version").read_text().strip()
+    return version
+
+
+def update_latest_version() -> None:
+    project_path = get_project_path()
+    if project_path is None:
+        return
+    url = "https://pypi.org/pypi/tach/json"
+    try:
+        # Sending a GET request to the URL
+        with request.urlopen(url, timeout=1) as response:
+            if response.status == 200:
+                data = response.read().decode()
+                json_data = json.loads(data)
+                latest_version = json_data["info"]["version"]
+            else:
+                return
+    except (error.URLError, KeyError):
```

Comment:
might also be good to catch errors related to JSON parsing or utf-8 decoding as well to avoid printing them to stdout when the tool exits (although they should be unlikely)